File: Creating a simple plugin mechanism (Part 2)/[VB]-Creating a simple plugin mechanism (Part 2)/VB/FirstPlugin/FirstPlugin.vb
Imports PluginContracts
Imports System.ComponentModel.Composition

<Export(GetType(IPlugin))>
Public Class FirstPlugin
    Implements IPlugin

    Public Sub DoSomething() Implements IPlugin.DoSomething
        System.Windows.MessageBox.Show("Do Something in First Plugin")
    End Sub

    Public ReadOnly Property Name As String Implements IPlugin.Name
        Get
            Return "First Plugin"
        End Get
    End Property
End Class

File: Creating a simple plugin mechanism (Part 2)/[C#]-Creating a simple plugin mechanism (Part 2)/C#/FirstPlugin/FirstPlugin.cs
using System.ComponentModel.Composition;
using PluginContracts;

namespace FirstPlugin
{
	[Export(typeof(IPlugin))]
	public class FirstPlugin : IPlugin
	{
		#region IPlugin Members

		public string Name
		{
			get
			{
				return "First Plugin";
			}
		}

		public void Do()
		{
			System.Windows.MessageBox.Show("Do Something in First Plugin");
		}

		#endregion
	}
}